forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/TaskRestResource.java

```diff
@@ -1,19 +1,95 @@
 
-import javax.ws.rs.core.Context;
-import javax.ws.rs.core.MediaType;
-import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+import java.util.List;
+import java.util.Map;
 
-@RequestMapping(RestResource.BASE_PATH+TaskRestResource.PATH)
+@Produces(MediaType.APPLICATION_JSON)
 public interface TaskRestResource extends RestResource {
 
+    public final static String DESERIALIZE_VALUE_QUERY_PARAM = "deserializeValue";
+    public final static String DESERIALIZE_VALUES_QUERY_PARAM = DESERIALIZE_VALUE_QUERY_PARAM + "s";
+
     String PATH = "/task";
 
-    @GetMapping(value = "/count", produces = MediaType.APPLICATION_JSON)
-    ResponseEntity<CountResultDto> getTasksCount(@Context UriInfo uriInfo);
+    @GET
+    @Produces({MediaType.APPLICATION_JSON, Hal.APPLICATION_HAL_JSON})
+    Object getTasks(@Context Request request, @Context UriInfo uriInfo,
+                                                          @QueryParam("firstResult") Integer firstResult, @QueryParam("maxResults") Integer maxResults);
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    List<TaskDto> queryTasks(TaskQueryDto query,
+                             @QueryParam("firstResult") Integer firstResult, @QueryParam("maxResults") Integer maxResults);
+    @GET
+    @Path("/count")
+    @Produces(MediaType.APPLICATION_JSON)
+    CountResultDto getTasksCount(@Context UriInfo uriInfo);
+
+    @POST
+    @Path("/count")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    CountResultDto queryTasksCount(TaskQueryDto query);
+
+    @GET
+    @Path("/{id}")
+    @Produces({MediaType.APPLICATION_JSON, Hal.APPLICATION_HAL_JSON})
+    Object getTask(@Context Request request, @PathParam("id") String id);
+
+    @PUT
+    @Path("/{id}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    void updateTask(TaskDto task, @PathParam("id") String id);
+
+    @POST
+    @Path("/{id}/claim")
+    @Consumes(MediaType.APPLICATION_JSON)
+    void claim(UserIdDto userIdDto, @PathParam("id") String id);
+
+    @POST
+    @Path("/{id}/unclaim")
+    void unClaim(@PathParam("id") String id);
+
+    @POST
+    @Path("/{id}/assignee")
+    @Consumes(MediaType.APPLICATION_JSON)
+    void setAssignee(UserIdDto dto, @PathParam("id") String id);
+
+    @GET
+    @Path("/{id}/variables")
+    @Produces(MediaType.APPLICATION_JSON)
+    Map<String, VariableValueDto> getVariables(@QueryParam(DESERIALIZE_VALUES_QUERY_PARAM) @DefaultValue("true") boolean deserializeValues, @PathParam("id") String id);
+
+    @GET
+    @Path("/{id}/identity-links")
+    @Produces(MediaType.APPLICATION_JSON)
+    List<IdentityLinkDto> getIdentityLinks(@QueryParam("type") String type, @PathParam("id") String id);
+
+    @POST
+    @Path("/{id}/identity-links")
+    @Consumes(MediaType.APPLICATION_JSON)
+    void addIdentityLink(IdentityLinkDto identityLink, @PathParam("id") String id);
```

Comment:
are we using these api's? please add only required apis

Reply:
updated